# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1762553859 -3600
# Parent  4fd6d6bfa9e3e26d7100b3989458dd57663377e6
Feature: Allow unsubscribe of no-select folders.

diff --git a/mailnews/imap/src/nsImapIncomingServer.cpp b/mailnews/imap/src/nsImapIncomingServer.cpp
--- a/mailnews/imap/src/nsImapIncomingServer.cpp
+++ b/mailnews/imap/src/nsImapIncomingServer.cpp
@@ -38,8 +38,9 @@
 #include "mozilla/Components.h"
 #include "nsNetUtil.h"
 #include "mozilla/Utf8.h"
 #include "mozilla/LoadInfo.h"
+#include "mozilla/Preferences.h"
 
 using namespace mozilla;
 using mozilla::net::LoadInfo;
 
@@ -949,8 +950,12 @@ NS_IMETHODIMP nsImapIncomingServer::Poss
       rv = rootFolder->FindSubFolder(dupFolderPath, getter_AddRefs(msgFolder));
       NS_ENSURE_SUCCESS(rv, rv);
       m_subscribeFolders.AppendObject(msgFolder);
       noSelect = (boxFlags & kNoselect) != 0;
+      if (Preferences::GetBool("mail.imap.allow_unsubscribe_noselect")) {
+        // Ignore that the folder was noSelect.
+        noSelect = false;
+      }
       nsCOMPtr<nsIMsgImapMailFolder> imapFolder =
           do_QueryInterface(msgFolder, &rv);
       NS_ENSURE_SUCCESS(rv, rv);
       imapFolder->SetHierarchyDelimiter(hierarchyDelimiter);
diff --git a/mailnews/mailnews.js b/mailnews/mailnews.js
--- a/mailnews/mailnews.js
+++ b/mailnews/mailnews.js
@@ -1166,8 +1166,10 @@ pref("mail.export.loglevel", "Warn");
 // When true, disk cache is used for messages not in offline store. If false,
 // memory cache is used instead. Both use the cache2 implementation.
 pref("mail.imap.use_disk_cache2", true);
 
+pref("mail.imap.allow_unsubscribe_noselect", false);
+
 #ifdef MOZ_THUNDERBIRD_RUST
   // Enable support for Microsoft Exchange via Exchange Web Services.
 #ifdef NIGHTLY_BUILD
 pref("experimental.mail.ews.enabled", true);
